packages/rfq-indexer/indexer/src/index.ts

```diff
@@ -51,147 +50,173 @@ ponder.on('FastBridgeV2:BridgeRequested', async ({ event, context }) => {
-        transactionHash: hash,
-      }
-    })
+  const {
+    args: { transactionId, relayer },
+    block: { timestamp },
+    transaction: { hash },
+    log: { address, blockNumber }, // may want to add address here eventually
+  } = event
+
+  const {
+    db: { BridgeProofProvidedEvents },
+    network: { chainId },
+  } = context
+
+  await BridgeProofProvidedEvents.upsert({
+    id: transactionId,
+    // Save the full data first time we index this event
+    create: {
+      transactionId,
+      relayer: trim(relayer),
+      originChainId: chainId,
+      originChain: getChainName(Number(chainId)),
+      blockNumber: BigInt(blockNumber),
+      blockTimestamp: Number(timestamp),
+      transactionHash: hash,
+    },
+    // Update the data with the latest event data on subsequent indexes
+    update: {
+      relayer: trim(relayer),
+      blockNumber: BigInt(blockNumber),
+      blockTimestamp: Number(timestamp),
+      transactionHash: hash,
+    },
+  })
 })
 
 ponder.on('FastBridgeV2:BridgeDepositClaimed', async ({ event, context }) => {
-    const {
-        args: { transactionId, relayer, to, token, amount },
-        block: { timestamp },
-        transaction: { hash },
-        log: { blockNumber },
-    } = event
-
-    const {
-        db: { BridgeDepositClaimedEvents },
-        network: { chainId },
-      } = context
-
-    await BridgeDepositClaimedEvents.create({
-      id: transactionId,
-      data: {
-        transactionId,
-        relayer: trim(relayer),
-        to: trim(to),
-        token: trim(token),
-        amount,
-        amountFormatted: formatAmount(amount, token),
-        originChainId: chainId, 
-        originChain: getChainName(Number(chainId)),
-        blockNumber: BigInt(blockNumber),
-        blockTimestamp: Number(timestamp),
-        transactionHash: hash,
-      },
-    })
+  const {
+    args: { transactionId, relayer, to, token, amount },
+    block: { timestamp },
+    transaction: { hash },
+    log: { blockNumber },
+  } = event
+
+  const {
+    db: { BridgeDepositClaimedEvents },
+    network: { chainId },
+  } = context
+
+  await BridgeDepositClaimedEvents.create({
+    id: transactionId,
+    data: {
+      transactionId,
+      relayer: trim(relayer),
+      to: trim(to),
+      token: trim(token),
+      amount,
+      amountFormatted: formatAmount(amount, token),
+      originChainId: chainId,
+      originChain: getChainName(Number(chainId)),
+      blockNumber: BigInt(blockNumber),
+      blockTimestamp: Number(timestamp),
+      transactionHash: hash,
+    },
+  })
 })
 
-/* DESTINATION CHAIN EVENTS */
+ponder.on('FastBridgeV2:BridgeProofDisputed', async ({ event, context }) => {
+  const {
+    args: { transactionId, relayer },
+    block: { timestamp },
+    transaction: { hash },
+    log: { blockNumber },
+  } = event
 
+  const {
+    db: { BridgeProofDisputedEvents },
+    network: { chainId },
+  } = context
+
+  await BridgeProofDisputedEvents.create({
```

Comment:
probably needs to be an upsert since multiple disputes are possible

Reply:
fixed